configure.ac

```diff
@@ -349,6 +391,64 @@ dnl remove this eventually.
 AC_DEFINE([USE_GMP], [1], [for backwards compatibility])
 
 
+dnl Find zlib
+AC_ARG_WITH([zlib],
+  [AS_HELP_STRING([--with-zlib@<:@=builtin|PREFIX@:>@],
+    [prefix of zlib installation. e.g. /usr/local; specify `builtin' to let GAP builds its own version of zlib])],
+    [],[with_zlib=yes])
+
+BUILD_ZLIB=no
+ZLIB_CPPFLAGS=
+ZLIB_LDFLAGS=
+ZLIB_LIBS="-lz"
+AS_CASE([$with_zlib],
+  [builtin],[
+    # user explicitly requested to use builtin zlib
+    BUILD_ZLIB=yes
+    AC_MSG_NOTICE([Using bundled zlib])
+  ],
+  [no],[
+    AC_MSG_ERROR([Building without zlib is not supported])
+  ],
+  [system],[with_zlib=yes],  dnl supported for backwards compatibility with old build system
+  [yes],[],
+  [*],[
+    ZLIB_CPPFLAGS="-I${with_zlib}/include"
+    ZLIB_LDFLAGS="-L${with_zlib}/lib"
+  ]
+)
+
+AS_IF([test $BUILD_ZLIB = no],
+  [
+    # try to link against zlib
+    AX_CHECK_LIBRARY([ZLIB], [zlib.h], [z], [inflateEnd])
+    AS_IF([test $ax_cv_have_ZLIB = no],[
+        AS_IF([test "x$with_zlib" = "xyes"],[
+          BUILD_ZLIB=yes
+          AC_MSG_NOTICE([No usable zlib found, switching to included zlib])
+        ],
+        [AC_MSG_ERROR([zlib not found at prefix $with_zlib])]
+        )
+    ])
+  ])
+
+# Use bundled zlib if requested
+AS_IF([test x$BUILD_ZLIB = xyes],[
+  AS_IF([test x$CYGWIN = xyes],[
```

Comment:
It seems a bit overkill to move the a huge section of `configure.ac` just to check for cygwin here, esp. as that only requires matching `$host_os` against `*cygwin*`...

Reply:
How do I do substring matching in configure scripts? Google isn't turning up anything useful.

Reply:
I eventually managed to track down you are supposed to use AC_CASE of course....

Reply:
Fixed